metatensor-core/src/io/block.rs

```diff
@@ -137,27 +137,60 @@ fn read_data<R, F>(mut reader: R, create_array: &F) -> Result<(mts_array_t, Vec<
         return Err(Error::Serialization("data can not be loaded from fortran-order arrays".into()));
     }
 
-    let shape = header.shape;
-    let mut array = create_array(shape.clone())?;
+    let shape = header.shape.clone();
+    let (array, shape) = read_data_dlpack(reader, create_array, shape)?;
 
-    match header.type_descriptor {
-        DataType::Scalar(s) if s == "<f8" => {
-            reader.read_f64_into::<LittleEndian>(array.data_mut()?)?;
-        }
-        DataType::Scalar(s) if s == ">f8" => {
-            reader.read_f64_into::<BigEndian>(array.data_mut()?)?;
-        }
-        _ => {
-            return Err(Error::Serialization(format!(
-                "unknown type for data array, expected 64-bit floating points, got {}",
-                header.type_descriptor
-            )));
+    Ok((array, shape))
+}
+
+/// Read a data array from the given reader using DLPack.
+fn read_data_dlpack<R, F>(
```

Comment:
I don't understand why we need this function. We are reading from a file and creating arrays with `create_array`, DLPack should not be involved anywhere. Although we might have to change `create_array` to take a dtype argument